modules/validations/crossing_ways.js

```diff
@@ -476,6 +476,28 @@ export function validationCrossingWays(context) {
                     if (allowsTunnel(selectedFeatureType) && !skipTunnelFix) {
                         fixes.push(makeAddBridgeOrTunnelFix('add_a_tunnel', 'temaki-tunnel', 'tunnel'));
                     }
+
+                    // special case: if
+                    //  (1) we're about to join these lines with a highway=crossing node; and
+                    //  (2) one of the lines is a sidewalk
+                    // then we will split the sidewalk and create a highway=crossing way
+                    const isSidewalk = (
+                        entities[0].tags.footway === 'sidewalk' ||
+                        entities[1].tags.footway === 'sidewalk' ||
+                        entities[0].tags.cycleway === 'sidewalk' ||
+                        entities[1].tags.cycleway === 'sidewalk'
```

Comment:
this part needs more discussion, currently it only applies to `footway=sidewalk` and `cycleway=sidewalk`. 

I think it would make sense to apply this autofix to any `highway=X` where `X=crossing` is a valid tag. This means it would apply to [`footway`](https://taginfo.osm.org/tags/footway=crossing), [`cycleway`](https://taginfo.osm.org/tags/cycleway=crossing), [`bridleway`](https://taginfo.osm.org/tags/bridleway=crossing), and [`path`](https://taginfo.osm.org/tags/path=crossing)